bin/api-docs/update-api-docs.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * External dependencies
+ */
+const { join, relative, resolve, sep, basename, dirname } = require( 'path' );
+const glob = require( 'fast-glob' );
+const execa = require( 'execa' );
+const { Transform } = require( 'stream' );
+const { readFile } = require( 'fs' ).promises;
+
+/**
+ * README file tokens, defined as a tuple of token identifier, source path.
+ *
+ * @typedef {[string,string]} WPReadmeFileTokens
+ */
+
+/**
+ * README file data, defined as a tuple of README file path, token details.
+ *
+ * @typedef {[string,WPReadmeFileTokens]} WPReadmeFileData
+ */
+
+/**
+ * Path to root project directory.
+ *
+ * @type {string}
+ */
+const ROOT_DIR = resolve( __dirname, '../..' );
+
+/**
+ * Path to packages directory.
+ *
+ * @type {string}
+ */
+const PACKAGES_DIR = resolve( ROOT_DIR, 'packages' );
+
+/**
+ * Path to data documentation directory.
+ *
+ * @type {string}
+ */
+const DATA_DOCS_DIR = resolve(
+	ROOT_DIR,
+	'docs/designers-developers/developers/data'
+);
+
+/**
+ * Pattern matching start token of a README file.
+ *
+ * @type {RegExp}
+ */
+const TOKEN_PATTERN = /<!-- START TOKEN\((.+?(?:\|(.+?))?)\) -->/g;
```

Comment:
Can we document what this RegExp is targeting? Perhaps adding two examples (one with matched path and other with default path) would be helpful.

Reply:
I [recently](https://github.com/aduth/dones/commit/012649c54cb37e041c2a124eb7649cf560c4c10b#diff-a73199f6b673a7742e2f8377e84da5d8R37-R43) became acquainted with [ES2018 named capture groups](https://2ality.com/2017/05/regexp-named-capture-groups.html). Maybe that could help make the pattern itself a little clearer to decipher? Alternatively, I've tried [other approaches at documenting capture groups](https://github.com/aduth/tannin/blob/cd1c7447843df7751c4abd1b92aee03fe56bfba4/packages/sprintf/index.js#L33-L38). Your suggestion of a couple simple examples is probably a good idea to include in any case.

Reply:
Added the two use cases this is trying to match at https://github.com/WordPress/gutenberg/pull/18840/commits/0475f24e8de08fece235547563b17c463f9c35a6 

Named capture groups look nice. Also, I applaud the effort in documenting the tannin regexp :clap: